bundles/binding/org.openhab.binding.isy/src/main/java/org/openhab/binding/isy/ISYActiveBinding.java

```diff
@@ -0,0 +1,718 @@
+
+	/**
+	 * The client used to send and receive messages from ISY.
+	 */
+	private InsteonClient insteonClient;
+
+	/**
+	 * Constructor, nothing to see here.
+	 */
+	public ISYActiveBinding() {
+	}
+
+	/**
+	 * Called by SCR when the binding is resolved
+	 * 
+	 * @param config
+	 */
+	public void activate(final Map<Object, Object> config) {
+
+		runtimeConfiguration = new ISYActiveBindingConfig(config);
+
+		setProperlyConfigured(true);
+
+		// create the Insteon client using their SDK
+		//
+		this.insteonClient = new InsteonClient(runtimeConfiguration.getUser(),
+				runtimeConfiguration.getPassword(), this);
+		Errors.addErrorListener(new ErrorHandler());
+
+		if (runtimeConfiguration.isUseUpnp()) {
+			this.logger.info("ISY: using UPNP to communicate with ISY");
+
+			this.insteonClient.start();
+		} else {
+			this.logger.info("ISY: using URL to communicate with ISY");
+
+			try {
+				this.insteonClient.start(runtimeConfiguration.getUuid(), String
+						.format("http://%s:%s",
+								runtimeConfiguration.getIpAddress(),
+								runtimeConfiguration.getPort()));
+			} catch (Exception e) {
+				throw new RuntimeException(e);
+			}
+		}
+	}
+
+	/**
+	 * Called by SCR when the binding configuration was modified
+	 */
+	public void modified(final Map<Object, Object> config) {
+
+		runtimeConfiguration = new ISYActiveBindingConfig(config);
+
+		setProperlyConfigured(true);
+
+		// create the Insteon client using their SDK
+		//
+
+		if (this.insteonClient != null) {
+			this.insteonClient.stop();
+		}
+
+		this.insteonClient = new InsteonClient(runtimeConfiguration.getUser(),
+				runtimeConfiguration.getPassword(), this);
+		Errors.addErrorListener(new ErrorHandler());
+
+		if (runtimeConfiguration.isUseUpnp()) {
+			this.logger.info("ISY: using UPNP to communicate with ISY");
+
+			this.insteonClient.start();
+		} else {
+			this.logger.info("ISY: using URL to communicate with ISY");
+
+			try {
+				this.insteonClient.start(runtimeConfiguration.getUuid(), String
+						.format("http://%s:%s",
+								runtimeConfiguration.getIpAddress(),
+								runtimeConfiguration.getPort()));
+			} catch (Exception e) {
+				throw new RuntimeException(e);
+			}
+		}
+	}
+
+	/**
+	 * Called by the SCR to deactivate the component when either the
+	 * configuration is removed or mandatory references are no longer satisfied
+	 * or the component has simply been stopped.
+	 */
+	@Override
+	public void deactivate() {
+		if (this.insteonClient != null) {
+			this.insteonClient.stop();
+			this.insteonClient = null;
+		}
+	}
+
+	/**
+	 * @{inheritDoc
+	 */
+	@Override
+	protected long getRefreshInterval() {
+		return this.runtimeConfiguration.getRefreshInterval();
+	}
+
+	/**
+	 * @{inheritDoc
+	 */
+	@Override
+	protected String getName() {
+		return "ISY Refresh Service";
+	}
+
+	/**
+	 * @{inheritDoc
+	 */
+	@Override
+	protected void execute() {
+		// the frequently executed code (polling) goes here ...
+		if (this.logger.isDebugEnabled()) {
```

Comment:
ok, I tend to do this for performance as a check is faster than the log check in most cases. Do not know this logger library well so will defer to you.